### PR TITLE
Check PV volumeMode in kubelet even when block feature is off

### PR DESCRIPTION
**Pull request by jsafrane**
/kind feature

**What this PR does / why we need it**:
Kubelet should check that a volume used in `pod.spec.containers[*].volumeMounts` is indeed filesystem volume and not block even when `BlockVolume` feature gate is disabled. The gate may be enabled in API server and/or controller-manager and they may already provision / bind block PV/PVCs and schedule pods that use them. If such a block volume is then used in pod `volumeMounts`, it would be formatted and mounted, corrupting raw data  on the volume.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/b1b2751/keps/sig-storage/20191008-raw-block-support.md
```

cc @msau42

### CODE REVIEW

**Comment by davidz627** [comment]
/lgtm
/retest
This should probably be cherry-picked to previous releases

File: pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go [hunk 1]
```diff
@@ -505,19 +505,21 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
 			pvcSource.ClaimName,
 			pvcUID)
 
-		// TODO: remove feature gate check after no longer needed
+		// TODO: remove features.BlockVolume checks / comments after no longer needed
+		volumeMode, err := util.GetVolumeMode(volumeSpec)
```
Comment (msau42) [inline pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: When feature gate is off, then VolumeMode field will be nil.  Then this function will return an error.

Comment (davidz627) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

Comment (jsafrane) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: > why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

The test is marked as [Slow]. 

https://github.com/kubernetes/kubernetes/blob/1bb7835f0aef486c9037cd83d0e59f7ee48c5551/test/e2e/storage/testsuites/volumemode.go#L287

Comment (jsafrane) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: Fixed. Good catch!

Comment (jsafrane) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: >> why didn't any pull jobs (unit/e2e tests) catch this? We should fix that

>The test is marked as [Slow].

It's more complicated... We run Slow tests in e2e. The reason it passes is that API server defaults volumeMode to Filesystem:

https://github.com/kubernetes/kubernetes/blob/5e8ccda71cc676e070cd0f5d7b6990b8ebc24c1f/pkg/apis/core/v1/defaults.go#L279-L282

(btw, it does so even when the BlockVolume feature gate is off).

I am not sure it's actually possible to get `volumeMode = nil`.

Comment (msau42) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go]: Rest strategy [drops fields](https://github.com/kubernetes/kubernetes/blob/5e8ccda71cc676e070cd0f5d7b6990b8ebc24c1f/pkg/api/persistentvolume/util.go#L27:6) when feature gate is off. Unsure if this comes into play when reading from a k8s client.

**Comment by msau42** [comment]
/lgtm
Thanks! Can you look into backporting this so we can cover the rollback case?

File: pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go [hunk 2]
```diff
@@ -567,6 +567,55 @@ func TestCreateVolumeSpec_Invalid_Block_VolumeMounts(t *testing.T) {
 	}
 }
 
+func TestCreateVolumeSpec_Invalid_Block_VolumeMounts_Disabled(t *testing.T) {
```
Comment (msau42) [inline pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go]: Oh actually, can you add one more test case for the nil volumeMode?

Comment (jsafrane) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go]: New test added.

**Pull request merged**